go/runtime/txpool/txpool.go

```diff
@@ -449,6 +451,32 @@ func (t *txPool) checkTxBatch(ctx context.Context, rr host.RichRuntime) {
 	checkCtx, cancel := context.WithTimeout(ctx, checkTxTimeout)
 	defer cancel()
 
+	// Ensure block round is synced to storage.
```

Comment:
How long can this block for? We should probably limit it.

An alternative would be doing checks against the latest block that we have synced, but this could be more out of date, not sure if this is a problem.

Reply:
I think it should be generally pretty fast/not even needed? Maybe using the `checkCtx` (which uses `checkTxTimeout`) is a good limit for this (I missed this context before, that's why it uses the parent one). Can also add a shorter limit if we prefer.

I think that before this waiting was implicitly done by the synced storage, so I think it shouldn't be a big difference having this here?

Reply:
Just using checkCtx sounds good.